omg/entities/projectile.py

```diff
@@ -27,58 +29,50 @@ def __init__(self, name, image_file, scale, damage, speed, init_px, init_py, ang
 
 
 class ProjectileFactory(ABC):
-    """Abstract class for Projectile creating factories."""
-
-    image_file: str
-
-    @classmethod
-    @abstractmethod
-    def create(cls, init_px: float, init_py: float, angle: float) -> Projectile:
-        """Create and return a Projectile instance."""
-        pass
-
-
-class FireballFactory(ProjectileFactory):
-    """Convenience class to create Fireball Projectile."""
-
-    image_file = os.path.join(ASSET_DIR, "skills", "fireball.PNG")
-    scale = 0.05
-    damage = 25
-    speed = 5
-
-    @classmethod
-    def create(cls, init_px, init_py, angle) -> Projectile:
-        """Create a Fireball projectile."""
+    """Abstract base class for projectile factories."""
+
+    def __init__(self) -> None:
+        self.name: str = None
+        self.image_file: str = None
+        self.scale: float = None
+        self.damage: float = None
+        self.speed: float = None
+        self.mana_cost: float = None
+
+    def create(self, init_px, init_py, angle) -> Projectile:
+        """Create a projectile with class-specific attributes."""
         return Projectile(
-            name=cls.__name__,
-            image_file=cls.image_file,
-            scale=cls.scale,
-            damage=cls.damage,
+            name=self.name,
+            image_file=self.image_file,
+            scale=self.scale,
+            damage=self.damage,
             init_px=init_px,
             init_py=init_py,
-            speed=cls.speed,
+            speed=self.speed,
             angle=angle,
         )
 
 
-class IceShardFactory(ProjectileFactory):
-    """Convenience class to create IceShard Projectile."""
+with open(CRAFTED_SKILLS_JSON_DIR, "r") as file:
+    crafted_skill_dictionary: Dict[str, Dict] = json.load(file)
 
-    image_file = os.path.join(ASSET_DIR, "skills", "ice_shard.PNG")
-    scale = 0.05
-    damage = 15
-    speed = 7
 
-    @classmethod
-    def create(cls, init_px, init_py, angle) -> Projectile:
-        """Create an IceShard projectile."""
-        return Projectile(
-            name=cls.__name__,
-            image_file=cls.image_file,
-            scale=cls.scale,
-            damage=cls.damage,
-            init_px=init_px,
-            init_py=init_py,
-            speed=cls.speed,
-            angle=angle,
-        )
+class SkillFactory(ProjectileFactory):
+    """Class to combine elements to craft a skill."""
+
+    def __init__(self) -> None:
+        self.name = None
+        self.image_file = None
+        self.scale = None
+        self.damage = None
+        self.speed = None
+        self.mana_cost = None
+
+    def set_skill_attributes(self, skill_name: str):
+        """Set the class attributes based on the skill_name."""
+        self.name = skill_name
+        self.image_file = crafted_skill_dictionary[skill_name]["image_file"]
```

Comment:
since crafted_skill_dictionary is a dict, you can use dict.get(key, default_value) method to not risk any key errors. I would also do it once, something like, `current_skill = crafted_skill_dictionary.get(skill_name, None)` and use `current_skill.image_file` etc below.